File: Retrofit/app/src/main/java/com/osos/retrofit/api/ApiClient.java
package com.osos.retrofit.api;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;

public class ApiClient
{
public static final String BASE_URl="http://api.openweathermap.org/";
public static Retrofit retrofit = null;
public static  Retrofit getRetrofit(){
    if(retrofit==null){
        retrofit= new Retrofit.Builder().baseUrl(BASE_URl).addConverterFactory(GsonConverterFactory.create()).build();
    }
    return retrofit;
}


}

File: android-search-example/app/src/main/java/com/osos/sliceassignment/api/ApiClient.kt
package com.osos.sliceassignment.api


import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {

    fun getInstance(): Retrofit {
        return Retrofit.Builder()
                .baseUrl(CONSTANTS.url)
                .addConverterFactory(GsonConverterFactory.create())
                .build() //Doesn't require the adapter
    }

    val apiCalls: ApiCalls = getInstance().create(ApiCalls::class.java)

}